realm/realm-library/src/androidTest/kotlin/io/realm/ManagedSetTester.kt

```diff
@@ -0,0 +1,280 @@
+        // TODO
+    }
+
+    override fun size() {
+        val set = initAndAssert()
+        assertEquals(0, set.size)
+        realm.executeTransaction {
+            initializedSet.forEach { value ->
+                set.add(value)
+            }
+        }
+        assertEquals(initializedSet.size, set.size)
+    }
+
+    override fun isEmpty() {
+        val set = initAndAssert()
+        assertTrue(set.isEmpty())
+        realm.executeTransaction {
+            initializedSet.forEach { value ->
+                set.add(value)
+            }
+        }
+        assertFalse(set.isEmpty())
+    }
+
+    override fun contains() {
+        val set = initAndAssert()
+        assertFalse(set.contains(notPresentValue))
+        realm.executeTransaction {
+            set.add(notPresentValue)
+        }
+        assertTrue(set.contains(notPresentValue))
+    }
+
+    override fun iterator() {
+        // TODO
+    }
+
+    override fun toArray() {
+        // TODO
+    }
+
+    override fun add() {
+        val set = initAndAssert()
+        assertEquals(0, set.size)
+        realm.executeTransaction {
+            // Adding a value for the first time returns true
+            initializedSet.forEach { value ->
+                assertTrue(set.add(value))
+            }
+            // Adding an existing value returns false
+            initializedSet.forEach { value ->
+                assertFalse(set.add(value))
+            }
+        }
+    }
+
+    override fun remove() {
+        val set = initAndAssert()
+        assertEquals(0, set.size)
+        realm.executeTransaction {
+            initializedSet.forEach { value ->
+                set.add(value)
+                assertTrue(set.remove(value))
+            }
+            assertFalse(set.remove(notPresentValue))
+        }
+        assertEquals(0, set.size)
+    }
+
+    override fun containsAll() {
+        // TODO
+    }
+
+    override fun addAll() {
+        // TODO
+    }
+
+    override fun retainAll() {
+        // TODO
+    }
+
+    override fun removeAll() {
+        // TODO
+    }
+
+    override fun clear() {
+        val set = initAndAssert()
+        realm.executeTransaction {
+            set.add(notPresentValue)
+            assertEquals(1, set.size)
+            set.clear()
+            assertEquals(0, set.size)
+        }
+    }
+
+    override fun freeze() {
+        // TODO
+    }
+
+    //----------------------------------
+    // Private stuff
+    //----------------------------------
+
+    private fun initAndAssert(realm: Realm = this.realm): RealmSet<T> {
+        val allTypesObject = createAllTypesManagedContainerAndAssert(realm)
+        assertNotNull(allTypesObject)
+        return setGetter.call(allTypesObject)
+    }
+}
+
+fun managedSetFactory(): List<SetTester> {
+    val primitiveTesters: List<SetTester> = SetSupportedType.values().mapNotNull { supportedType ->
+        when (supportedType) {
+//            SetSupportedType.LONG ->
```

Comment:
Maybe separate the `SetSupportedType.values()` into a local variable that you could initialize with single elements for test purpose. Seems way faster that selectively commenting stuff in and out.

Reply:
This will be addressed in https://github.com/realm/realm-java/issues/7362